src/serial.rs

```diff
@@ -7,40 +7,15 @@ use crate::{
     rcc::{Clocks, APB1, APB2},
     time::Bps,
 };
+use cfg_if::cfg_if;
 use core::{convert::Infallible, marker::PhantomData, ptr};
 
-#[cfg(any(
-    feature = "stm32f302",
-    feature = "stm32f303xb",
-    feature = "stm32f303xc",
-    feature = "stm32f303xd",
-    feature = "stm32f303xe",
-    feature = "stm32f373",
-    feature = "stm32f378",
-    feature = "stm32f334",
-    feature = "stm32f328",
-    feature = "stm32f358",
-    feature = "stm32f398"
-))]
-use crate::gpio::gpiod;
-#[cfg(any(
-    feature = "stm32f302",
-    feature = "stm32f303xb",
-    feature = "stm32f303xc",
-    feature = "stm32f303xd",
-    feature = "stm32f303xe",
-    feature = "stm32f373",
-    feature = "stm32f378",
-    feature = "stm32f328",
-    feature = "stm32f358",
-    feature = "stm32f398"
-))]
-use crate::gpio::gpioe;
-
-#[cfg(feature = "stm32f303")]
-use crate::dma;
-#[cfg(feature = "stm32f303")]
-use cortex_m::interrupt;
+cfg_if! {
+    if #[cfg(feature = "stm32f303")] {
+        use crate::dma;
+        use cortex_m::interrupt;
+    }
+}
```

Comment:
I am not completely happy with using `cfg_if` to apply feature gates to multiple items. I guess it beats repeating yourself, but there is quite a lot of added indentation now.